plugins/pencilblue/controllers/actions/salesforce/login_callback.js

```diff
@@ -38,10 +38,33 @@ module.exports = function LoginSalesforceCallbackControllerModule(pb) {
                     pb.log.warn('Something went wrong during the state parsing: ', e);
                 }
             }
-            let redirectLocation = await this.getRedirectLink(response, state);
+            if (response.updateEmail) {
+                await this.updateEmail(response.user);
+            }
+            let redirectLocation = await this.getRedirectLink(response.user, state);
             this.redirect(redirectLocation, cb);
         }
 
+        async updateEmail(user) {
+            try {
+                this.identityProviderService = this.createService('IdentityProviderService', 'tn_auth');
+                this.encryptProfileIdService = this.createService('EncryptProfileIdService', 'tn_profile');
```

Comment:
You need to make sure tn_profile is installed for this site before attempting to create a service from it:
if (pb.PluginService.isPluginActiveBySite('tn_profile', this.site)) {